src/core/compiler.h

```diff
@@ -0,0 +1,23 @@
+/**
+ * @file core/compiler.h
+ *
+ * @brief Compiler version detection variable
+ *
+ * Compiler version detection variable
+ *
+ * The variable rootsim_compiler_version is used to detect the compiler version.
+ * It is set to 0 by default, but it is overridden by the compiler when a model is compiled.
+ * The macro ROOTSIM_COMPILER_VERSION is used to check the compiler version.
+ *
+ * SPDX-FileCopyrightText: 2008-2023 HPDCS Group <rootsim@googlegroups.com>
+ * SPDX-License-Identifier: GPL-3.0-only
+ */
+#pragma once
+
+#include <stdint.h>
+
+/// The version of the compiler used to generate a model
+uint32_t __attribute__((weak)) rootsim_compiler_version = 0;
```

Comment:
We could _declare_ this variable here rather than _defining_ it inside this header, and define it as a weak symbol inside ```core.c```